### ISSUE TITLE
[ALIASES] Aliased column names are being used in filter queries

### ISSUE THREAD

**Issue by gfiorav** [cartodb-org/cartodb#160]
## STR

0. Have a dataset with aliased column names
1. Use a wizard (edit mode) and choose a column with an aliased name.

## Expected result

The wizard works

## Actual result

There's an SQL error. The source of the error is that the query created for the filter contains aliased column names in it. 

Any ideas @coyle5280 ?

**Comment by coyle5280** [comment]
Not sure off the top of my head. Is this an issue on the source branch or an issue with my last pull request?

**Comment by gfiorav** [comment]
Hey @coyle5280,

This is an issue with the aliases feature. When you add a filter, what happens in the background is that the query is changed to show what the filter represents. So, for example:

1. User chooses a mix, max filter (histogram)
2. An SQL is generated as `SELECT min(4, column_1), max(8, column_1) FROM table;`
3. The filtered data is show in the map

The issue here is that the component that is creating this query in step 2, is taking the aliased names when available. Of course, the database knows nothing about alias and can't resolve the names, thus generating an error.

This is probably conflicting with the decision we made to "give the user the alias as if it were the name to reduce complexity". We overlooked that the frontend generates these queries (not ideal) and thus needs more info that the user handles (again, not ideal).

The solution would stream along the lines of: making sure the component that generates this query uses the original column names. For that they need to be made available. So, two options:

1. Go back to passing both alias and original name.

Pros: Simpler to handle at front, no need to modify query generator component.
Cons: Added complexity when displaying column. Passing info that the user doesn't care about and shouldn't need.

2. Stick to current implementation and add original name params. 

Pros: We don't need to change the column name stuff.
Cons: We need to modify the query generator component, making it lest robust against future updates (diverting from vanilla master version)


ATM, I'd go with 1. This means that we'd pass alias and original names, composing the "aliased column name" when needed, but preserving the column name logic. That is: the column names should be contained in attributes named the same as the original implementation (so everything, including the query component, knows where to look). I.e: instead of passing `original_name: 'column_name'`, add `alias: 'blabla'`.

It's going back, I know, but we're learning :)

This is my take on it, I'd love to hear yours!
